From e1a3e4b3bda6ecdb5f39170e0f88eaec8e85c05b Mon Sep 17 00:00:00 2001
From: "haotian.qu" <haotian.qu@amlogic.com>
Date: Thu, 9 Nov 2023 17:41:03 +0800
Subject: [PATCH 1/2] aml_swupdate_ui: Compile output to the build directory.
 [2/3]

PD#SWPL-145434

Problem:
The compiled files should be placed in the build directory.

Solution:
Add a new output directory param to depositing the
compiled files

Verify:
local PB

Signed-off-by: haotian.qu <haotian.qu@amlogic.com>
Change-Id: Ibbd0483335d4e1ec555d6b025425e103bf550420
---
 Makefile              | 25 ++++++++++++++-----------
 common/Makefile       | 10 ++++------
 common/event_ui.c     |  4 ++--
 lvgl_porting/Makefile |  9 ++++++---
 lvgl_ui/Makefile      |  8 ++++----
 5 files changed, 30 insertions(+), 26 deletions(-)

diff --git a/Makefile b/Makefile
index 31166a9..d73eb2f 100644
--- a/Makefile
+++ b/Makefile
@@ -1,17 +1,20 @@
 .PHONYH: clean
+CFLAGS += -g -O3
 
-all: swupdateui
+OUT_DIR ?= $(CURDIR)
+
+all: $(OUT_DIR)/swupdateui
 
 ifeq ($(CONFIG_LVGL_APP), y)
-LDFLAGS += -llvgl -llv_drivers
-swupdateui: main.o lvgl_ui/liblvgl_ui.a lvgl_porting/liblvgl_porting.a common/libcommon.a
-	$(CXX) $^ -lpthread -o $@ $(LDFLAGS) -lswupdate
+LDFLAGS += -llvgl -llv_drivers -lswupdate
+$(OUT_DIR)/swupdateui: $(OUT_DIR)/main.o $(OUT_DIR)/liblvgl_ui.a $(OUT_DIR)/liblvgl_porting.a $(OUT_DIR)/libcommon.a
+	$(CXX) $^ -lpthread $(LDFLAGS) -o $@
 
-lvgl_ui/liblvgl_ui.a:
-	$(MAKE) -C lvgl_ui
+$(OUT_DIR)/liblvgl_ui.a:
+	$(MAKE) -C lvgl_ui OUTPUT=$(OUT_DIR)
 
-lvgl_porting/liblvgl_porting.a:
-	$(MAKE) -C lvgl_porting
+$(OUT_DIR)/liblvgl_porting.a:
+	$(MAKE) -C lvgl_porting OUTPUT=$(OUT_DIR)
 
 else
 LDFLAGS += $(shell $(PKG_CONFIG) --libs directfb) -l++dfb
@@ -23,10 +26,10 @@ directfb_ui/libdirectfb_ui.a:
 
 endif
 
-common/libcommon.a:
-	$(MAKE) -C common
+$(OUT_DIR)/libcommon.a:
+	$(MAKE) -C common OUTPUT=$(OUT_DIR)
 
-main.o: main.cpp
+$(OUT_DIR)/main.o: main.cpp
 	$(CXX) $(CFLAGS) -O3 $(LDFLAGS) -c $< -o $@
 
 clean:
diff --git a/common/Makefile b/common/Makefile
index 88733b7..ae226be 100644
--- a/common/Makefile
+++ b/common/Makefile
@@ -3,15 +3,13 @@
 #
 .PHONYH: clean
 
-OBJ=event_ui.o
+all: $(OUTPUT)/libcommon.a
 
-all: libcommon.a
-
-libcommon.a:$(OBJ)
+$(OUTPUT)/libcommon.a:$(OUTPUT)/event_ui.o
 	$(AR) -cvr $@ $^
 
-$(OBJ):%.o:%.c
-	$(CC) $(CFLAGS) -O3 $(LDFLAGS) -g -c $^ -o $@
+$(OUTPUT)/event_ui.o:event_ui.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -c $^ -o $@
 
 clean:
 	rm -rf *.o *.a
diff --git a/common/event_ui.c b/common/event_ui.c
index ec30418..0ea4d0f 100644
--- a/common/event_ui.c
+++ b/common/event_ui.c
@@ -79,8 +79,8 @@ static int calc_perc_of_total(struct progress_msg *msg)
     static int cur_step_base;
     int cur_perc_convert = 0;
     int total_percent = 0;
-    static last_cur_step;
-    static last_total_percent;
+    static int last_cur_step;
+    static int last_total_percent;
 
     /* Calculate how much of each step is a percentage of the total,
      * Each step is a stage, and the percentage of completed steps plus
diff --git a/lvgl_porting/Makefile b/lvgl_porting/Makefile
index 08e1ca2..f8637df 100644
--- a/lvgl_porting/Makefile
+++ b/lvgl_porting/Makefile
@@ -1,11 +1,14 @@
 .PHONYH: all install clean
 
-all: liblvgl_porting.a
+all: $(OUTPUT)/liblvgl_porting.a
 
-liblvgl_porting.a:lv_port_disp.o lv_port_fs.o
+$(OUTPUT)/liblvgl_porting.a:$(OUTPUT)/lv_port_disp.o $(OUTPUT)/lv_port_fs.o
 	$(AR) -cvr $@ $^
 
-lv_port_disp.o lv_port_fs.o:%.o:%.c
+$(OUTPUT)/lv_port_disp.o: lv_port_disp.c
+	$(CC) $(CFLAGS) -O3 $(LDFLAGS) -g -c $^ -o $@
+
+$(OUTPUT)/lv_port_fs.o: lv_port_fs.c
 	$(CC) $(CFLAGS) -O3 $(LDFLAGS) -g -c $^ -o $@
 
 clean:
diff --git a/lvgl_ui/Makefile b/lvgl_ui/Makefile
index 9df4d23..4bc4590 100644
--- a/lvgl_ui/Makefile
+++ b/lvgl_ui/Makefile
@@ -1,12 +1,12 @@
 .PHONYH: clean
 
-all: liblvgl_ui.a
+all: $(OUTPUT)/liblvgl_ui.a
 
-liblvgl_ui.a:aml_lvgl_ui.o
+$(OUTPUT)/liblvgl_ui.a:$(OUTPUT)/aml_lvgl_ui.o
 	$(AR) -cvr $@ $^
 
-aml_lvgl_ui.o:%.o:%.c
-	$(CC) $(CFLAGS) -O3 $(LDFLAGS) -g -c $^ -o $@
+$(OUTPUT)/aml_lvgl_ui.o:aml_lvgl_ui.c
+	$(CC) $(CFLAGS) $(LDFLAGS) -c $^ -o $@
 
 clean:
 	rm -rf *.o *.a
-- 
2.25.1

